source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -183,12 +187,16 @@ ParseState Filter::parseClientHello(const void* data, size_t len,
   ParseState state = [this, ret]() {
     switch (SSL_get_error(ssl_.get(), ret)) {
     case SSL_ERROR_WANT_READ:
-      if (read_ == config_->maxClientHelloSize()) {
+      if (read_ == maxConfigReadBytes()) {
         // We've hit the specified size limit. This is an unreasonably large ClientHello;
         // indicate failure.
         config_->stats().client_hello_too_large_.inc();
         return ParseState::Error;
       }
+      if (read_ == requested_read_bytes_) {
+        // Double requested bytes up to the maximum configured.
+        requested_read_bytes_ = std::min<uint32_t>(2 * requested_read_bytes_, maxConfigReadBytes());
```

Comment:
Is it worth adding a stat for helping the user to find out which best initial request read bytes? Or I'm asking how the user figures out which best initial request read bytes for him